cmd/asset-syncer/sync.go

```diff
@@ -58,6 +61,17 @@ var syncCmd = &cobra.Command{
 		}
 
 		authorizationHeader := os.Getenv("AUTHORIZATION_HEADER")
+		// The auth header may be a dockerconfig that we need to parse
+		dockerConfig := &credentialprovider.DockerConfigJson{}
+		err = json.Unmarshal([]byte(authorizationHeader), dockerConfig)
+		if err != nil {
+			// Not an dockerconfig, skip
```

Comment:
In the (distant) future when a new docker config format is enabled, we will error here without any knowledge of why... If possible, can we avoid ignoring an error?

For example, can we reverse the check here and see if `authorizationHeader` is a valid token, if not, check if it's a valid docker config, and if not, actually return the appropriate error? Hmm, though that doesn't work if the normal auth header is just an opaque token (which I think it is, from memory).

In which case, the sync-job *could* have a separate option for the docker config (which would be better than re-using `AUTHORIZATION_HEADER` anyway, I think?). If authorization header is set, it is used. If docker config is set, it is used (or we error if it can't be parsed).

Reply:
> In which case, the sync-job _could_ have a separate option for the docker config (which would be better than re-using `AUTHORIZATION_HEADER` anyway, I think?). If authorization header is set, it is used. If docker config is set, it is used (or we error if it can't be parsed).

Yep, probably it's better to do that (but need some additional changes in the controller).